### ISSUE TITLE
remove_from_group error when quitting game

### ISSUE THREAD

**Issue by volzhs** [godotengine/godot#45332]
**Godot version:**

3.2.4 6d5daac3e4

**OS/device including version:**

not related

**Issue description:**

Error appears when quitting game.
More precisely, this error occurred when `WorldEnvironment` node is being removed.
```
ERROR: remove_from_group: Condition "!data.grouped.has(p_identifier)" is true.
   At: scene/main/node.cpp:1698.
```
it's a regression from 4954a869bb.
before this commit, this does not happen.

**Steps to reproduce:**
1. Add `WorldEnvironment` node in scene
2. Run game
3. Quit

```
Node::remove_from_group(Node * const this, const StringName & p_identifier) (scene/main/node.cpp:1697)
WorldEnvironment::_notification(WorldEnvironment * const this, int p_what) (scene/3d/world_environment.cpp:50)
WorldEnvironment::_notificationv(WorldEnvironment * const this, int p_notification, bool p_reversed) (scene/3d/world_environment.h:38)
Object::notification(Object * const this, int p_notification, bool p_reversed) (core/object.cpp:929)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:305)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:294)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:294)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:294)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:294)
Node::_propagate_exit_tree(Node * const this) (scene/main/node.cpp:294)
Node::_set_tree(Node * const this, SceneTree * p_tree) (scene/main/node.cpp:2546)
SceneTree::finish(SceneTree * const this) (scene/main/scene_tree.cpp:619)
OS_X11::run(OS_X11 * const this) (platform/x11/os_x11.cpp:3611)
main(int argc, char ** argv) (platform/x11/godot_x11.cpp:56)
```

**Minimal reproduction project:**

[fxaa_transparent_bg.zip](https://github.com/godotengine/godot/files/5844252/fxaa_transparent_bg.zip)

**Comment by volzhs** [comment]
cc @lawnjelly

**Comment by lawnjelly** [comment]
Is probably an order of update thing, as the collision pairing is now being updated from draw, so it's possible some things are never getting unpaired during shutdown. Will take a look. :+1:

**Comment by lawnjelly** [comment]
So it turns out the function I was calling to loop through the `scenarios` in visual_server seems to be debug only. So I'm going to try and get @akien-mga to revert the PR temporarily, until I've consulted with reduz on the proper way to get the scenarios. Am too tired to address this tonight.

In the meantime you can turn the BVH off in `rendering->quality->spatial_partitioning->use_bvh` and it should work as normal.

**Comment by akien-mga** [comment]
Fixed by #45335.
